java/src/main/java/com/lancedb/lance/WriteParams.java

```diff
@@ -76,6 +88,10 @@ public Optional<Boolean> getEnableStableRowIds() {
     return enableStableRowIds;
   }
 
+  public Optional<String> getDataStorageVersion() {
+    return dataStorageVersion.map(Enum::name);
```

Comment:
I think this has to match the version string in rust https://github.com/lancedb/lance/blob/3df63ea7cfa7ed3ab63737bb47e58fca5688cf69/rust/lance-encoding/src/version.rs#L9-L12